program/Project Descriptions and Plans/CORE-V-HAL/Project-Concept-HAL.md

```diff
@@ -0,0 +1,155 @@
+
+It will be a key component of a Core-V MCU SDK.
+
+ 
+## Initial Estimate of Timeline
+The project will be last for 3 Months and have below output at different stage.
+
+- To agree the key technical requirements, approach, and specification for a Core-V HAL that will be re-used for all Core-V Cores
+
+Once the HAL approach is agreed
+   - To implement basic peripherals (UART, TIME, INT...) on Core-V MCU.
+   - To have FreeRTOS run  on Core-V MCU in conjunction with HAL SDK support 
+   - To implement support other peripherals and meet application requirement.
+
+## Explanation of why OpenHW should do this project
+
+- The HAL is needed to establish a baseline approach for addressing the Core-V Cores and peripherals under a consistent approach. It is a cornerstone of all Core-V SDK and software work.
+
+- There are some challenges for Core-V MCU users to develop application bases on adopted PULP SDK due to the complexity of the development configuration environment. The PULP SDK is not viewed as a basis for future development work. 
+
+- There should be an out-of-box SDK solution offered by OpenHW supporting Core-V MCU platform developer.
+
+- An easy-to-use SDK enhances the Core-V MCU development environment by reducing the barrier on software development and enabling to create reusable application code.
+
+- There are some requirement from system tools (such as compiler) to help verify their design's performance and correctness using HAL SDK.
+
+## Industry landscape: description of competing, alternative, or related efforts in the industry
+There are also some similar standards in market currently, the most representative HAL is below:
+   1. CMSIS for ARM cortex A and M architecture
+   2. NMSIS for RISC-V architecture
+   3. CSI for RISC-V architecture
+   4. FreeRTOS CommonIO specification https://developer.amazon.com/acs-devices
+
+### Table showing how existing solutions map against requirements at Project Concept gate
+
+| Items | CMSIS | NMSIS | CSI | CommonIO | 
+| --- | --- | --- | --- | --- |
+| Core | Support | Support | Support | Support |
+| Driver | Device x13 | - | Device x 23 | |
+| RTOS | FreeRTOS | - | FreeRTOS, Rhino | |
+| NN | Support | Support | Support | |
+| DSP | Support | Support | Support | |
+| Coding Rule | MISRA | MISRA | MISRA, TUeV61508 | |
+| Validation | Support | Support | Support | |
+| License | Apache 2.0 | Apache 2.0 | Apache 2.0 | |
+| Software pack | Support | Support | Support | |
+| **Numbered Requirements**| Y/N/Maybe |  |  | |
+| T1 |  |  |  |  |
+| T2 |  |  |  |  |
+| T3 |  |  |  |  |
+| T4 |  |  |  |  |
+| T5 |  |  |  |  |
+| T6 |  |  |  |  |
+| T7 |  |  |  |  |
+| T8 |  |  |  |  |
+| T9 |  |  |  |  |
+| IA1 |  |  |  |  |
+| S1 |  |  |  |  |
+| S2 |  |  |  |  |
+| S3 |  |  |  |  |
+| S4 |  |  |  |  |
+| IM1 |  |  |  |  |
+| IM2 |  |  |  |  |
+| IM3 |  |  |  |  |
+| IM4 |  |  |  |  |
+
```

Comment:
We have to use the full requirement names - it is unreadable to flick to and fro. And we remove all the original table rows.  For things that are not in the requirements list, we can add a row at the bottom for "additional features which are not requirements".